lib/fs/metadata.go

```diff
@@ -125,7 +128,7 @@ func (m Metadata) Marshal(out io.Writer) {
 	enc.EncodeString(magic_UTF8, "t") // type
 	enc.EncodeInt(int64(m.Typeflag))
 	enc.EncodeString(magic_UTF8, "m") // mode -- note this is *not* `os.FileMode`, it's just the perm bits
-	enc.EncodeInt(m.Mode)
+	enc.EncodeInt(m.Mode & 07777)
```

Comment:
Leave a comment explaining why the whitelist. Incidentally, why the leftmost `7` - my assumption was that `1777` is as far as magic bits go; knowing linux I'm probably insufficiently informed.